configure

```diff
@@ -0,0 +1,338 @@
+        rm testfile.o
+        sub_info Yes.
+        eval "$1=\"\${$1} \$2\""
+        eval "$1=\${$1# }"
+    else
+        if [ "$2" = "-std=c++11" ]; then
+            sub_error "Seems like C++ compiler don't support $2 but It's required!"
+        else
+            sub_info No.
+        fi
+    fi
+}
+
+try_ld_argument()
+{
+    info Checking whatever linker accepts "$2"...
+    if $compiler $CXXOPTS ${POST_CXXOPTS:-} $LDFLAGS ${POST_LDFLAGS:-} $2 testfile.cc -o testfile.o > /dev/null 2>&1; then
+        sub_info Yes.
+        eval "$1=\"\${$1} \$2\""
+        eval "$1=\${$1# }"
+    else
+        sub_info No.
+    fi
+}
+
+usage()
+{
+    cat << EOF
+Usage: $0 [OPTION]...
+
+Defaults for the options are specified in brackets.
+
+  --help                        This help message.
+  --quiet                       Don't print normal messages, just errors.
+  --clean                       Clear mconfig and configure's temp files.
+
+Target options:
+  --platform=PLATFORM           Set the platform manually (Just for cross-platform cross-compile!) [autodetected]
+                                  For all cross-compiles (even this) don't forget to set correct CXX and HOSTCXX!
+
+Installation directories:
+  --prefix=PREFIX               Main installtion prefix [/usr]
+  --exec-prefix=EPREFIX         Main executables location [/]
+  --sbindir=SBINDIR             Dinit executables [EPREFIX/sbin]
+  --mandir=MANDIR               Dinit man-pages location [PREFIX/share/man]
+  --syscontrolsocket=SOCKETPATH Dinitctl socket location [/run/dinitctl]
+
+Optional options:
+  --shutdown-prefix=PREFIX      Name prefix for shutdown, poweroff, reboot, halt programs []
+  --enable-shutdown             Build shutdown, poweroff, reboot, halt programs [Enabled only on Linux based systems]
+  --disable-shutdown            Don't build shutdown, poweroff, reboot, halt programs
+  --enable-cgroups              Enable Cgroups support [Enabled only on Linux based systems]
+  --disable-cgroups             Disable Cgroups support
+  --enable-sanitize             Enable -fsanitize for Unit tests [auto]
+  --disable-sanitize            Disable -fsanitize for Unit tests
+  --enable-utmpx                Enable manipulating the utmp/utmpx database via the related POSIX functions [auto]
+  --disable-utmpx               Disable manipulating the utmp/utmpx database via the related POSIX functions
+  --disable-cxxopts             (Not recommanded) Disable ALL CXXOPTS even -std=c++11
+                                  Don't forget set your flags via CXXOPTS_POST env otherwise
+                                  build may be failed!
+  --disable-ldflags             (Not recommanded) Disable ALL LDFLAGS even "-lrt on FreeBSD"
+                                  You can use LDFLAGS_POST env to set your ld flags.
+
+Environment variables can be used:
+  Note: CXXOPTS and LDFLAGS are Default compiler/linker flags. Don't use them. because
```

Comment:
This is confusing still. I think you should either remove all reference to CXXOPTS and LDFLAGS from the output, or allow them to be overridden just like other variables (and then you can get rid of the confusing `--disable-cxxopts`/`--disable-ldflags` options).

Saying they are "default compiler/linker flags" is ambiguous. It's better to be specific: they will be set to the options that the configure script uses as default for the platform, filtered to remove any options not supported by the compiler.

Reply:
I think if users need to avoid those arguments, just pass a `CXXOPTS` ( or `LDFLAGS`) var with empty value.
```
$ ./configure CXXOPTS=""
```
I working on it. but with `set -u` it is more difficult to do.

Reply:
I do this thing. Now users can replace `CXXOPTS` and `LDFLAGS` without any "confiusing option". Also if users want to fully disable `CXXOPTS` or `LDFLAGS` just pass `CXXOPTS=""` in arguments or pass `CXXOPTS="null"` as environment variable.
It's ok?

Reply:
> or pass CXXOPTS="null" as environment variable.

Why not just `CXXOPTS=""` regardless of how it is passed (environment or argument)?

Reply:
because I don't know how we can detect `CXXOPTS` is user defined variable (and have empty value) or it's just not exist! Do you have a solution for this?

Reply:
Try searching for shell parameter expansions rules.

`${VARIABLE+IS_SET}` will expand to `IS_SET` if the variable is set (including if set to ""), or the empty string ("") otherwise, for example.